apps/script/HoC2023AiGenerateWeights.py

```diff
@@ -101,61 +112,71 @@
-
-palette_dict = {}
-background_dict = {}
-foreground_dict = {}
-
-# Calculate and print similarity scores
-for emoji_name in emoji_names_list:
-    palette_scores = []
-    for palette_word in palette_model_descriptive_names:
-        id_token = nlp(emoji_name)
-        palette_token = nlp(palette_word)
-        similarity_score = id_token.similarity(palette_token)
-        palette_scores.append(round(similarity_score, 2))
-
-    background_scores = []
-    for background_word in background_effect_model_descriptive_names:
-        id_token = nlp(emoji_name)
-        background_token = nlp(background_word)
-        similarity_score = id_token.similarity(background_token)
-        background_scores.append(round(similarity_score, 2))
-
-    foreground_scores = []
-    for foreground_word in foreground_effect_model_descriptive_names:
-        id_token = nlp(emoji_name)
-        foreground_token = nlp(foreground_word)
-        similarity_score = id_token.similarity(foreground_token)
-        foreground_scores.append(round(similarity_score, 2))
-
-    emoji_id = emojis_map[emoji_name]
-    palette_dict[emoji_id] = palette_scores
-    background_dict[emoji_id] = background_scores
-    foreground_dict[emoji_id] = foreground_scores
-
-background_output = {'emojiAssociations': background_dict, 'output': background_effect_model_descriptive_names}
-palette_output = {'emojiAssociations': palette_dict, 'output': palette_model_descriptive_names}
-foreground_output = {'emojiAssociations': foreground_dict, 'output': foreground_effect_model_descriptive_names}
-
-background_output['pythonNames'] = background_output['output']
-palette_output['pythonNames'] = palette_output['output']
-foreground_output['pythonNames'] = foreground_output['output']
-
-# Modify output to reflect blockly id names.
-background_output['output'] = [background_effects_map[bg] for bg in background_output['output']]
-foreground_output['output'] = [foreground_effects_map[fg] for fg in foreground_output['output']]
-palette_output['output'] = [palettes_map[pal] for pal in palette_output['output']]
-
-with open("apps/static/dance/ai/model/cached-spacy-palette-map.json", "w") as json_file:
-    json_file.write(json.dumps(palette_output))
-
-with open("apps/static/dance/ai/model/cached-spacy-background-map.json", "w") as json_file:
-    json_file.write(json.dumps(background_output))
-
-with open("apps/static/dance/ai/model/cached-spacy-foreground-map.json", "w") as json_file:
-    json_file.write(json.dumps(foreground_output))
+
+# Create maps of model_descriptive_name to blockly_id and print lists
+# as 'model_descriptive_name | blockly_id | blockly_user_facing_name'
+# Updates lists in comments at bottom of this file.
+create_map_print_options(PALETTES, model_descriptive_names_map[PALETTES], palette_blockly_ids_list, palette_user_facing_names_list),
```

Comment:
Probably a Python thing I don't understand...what do the commas at the end of these lines represent?

Reply:
Hmmm... For me this is just a typo and I'm curious why I was still able to run script without an error message.  I'll go ahead and remove. 
I do see that trailing commas are allowed in lists/tuples in python which makes sense - but not in this case. 🤷‍♀️

Reply:
From reading python documentation, apparently the commas prevent a newline character from being added at the end of a print statement which seems to be the primary purpose of this helper function; I've personally never used this feature myself before LOL

Reply:
Interesting! Thanks!